### DISCUSSION TITLE
Problème signature de commit

### DISCUSSION THREAD

**Discussion by Syseria** [heig-vd-dai-course/heig-vd-dai-course#439] [Q&A] [answered]
Bonjour @ludelafo et @hadrylouis,

Suite à mes petits problème de PC, j'ai refait la mise en place de l'environnement mais sous Windows le temps de récupérer mon PC.
Un de points important pour le labo1 que nous devons vous rendre ce soir est la protection de la branch principale qui nécessite de signer nos commits comme stipulé dans le cours.
Le problème est que mon PC ne veut pas le faire... J'ai configuré une clé de signature SSH pour WSL (et une pour Windows directement, au cas où) et l'ai ajouté à GitHub, le tout comme indiqué dans la doc GitHub.
Impossible de signer les commits.

Suppression des clés SSH, régénération et reconfiguration -> toujours pas possible

J'ai tenté de signer les anciens commits comme indiqué [ici](https://medium.com/@aamir.shehzad3346875/how-to-sign-previous-commit-that-have-already-been-pushed-4683a7060e19), un seul a pu être signé rétroactivement (les clés fonctionnent donc) mais les suivants non pas été signé. Même après avoir réeffectuer la manipulation.

Est-il possible d'effectuer un merge de la branche dans le main localement et de la push sans avoir besoin de passer par une PR extraordinairement (si besoin désactiver la protection temporairement) ?

Notre projet est fini et fonctionnel, juste impossible de mettre à jour le main du à ce petit problème de signature.

Je récupère mon laptop Lundi et devrait pouvoir retravailler dans des conditions acceptable rapidement.

Vous pouvez vérifier l'avancé et le bon fonctionnement de notre application dans la branche concernée [ici](https://github.com/Bananonymous/DAI-Labo1/tree/Commands).

**Comment by hadrylouis** [comment, marked as answer]
Salut @Syseria, 

C'est en ordre pour moi si tous les commits ne sont pas signés pour cause de problème technique avec ton PC. 

Je te laisse faire le nécessaire pour que tout soit sur la branche main, si possible via une PR et si pas possible via un push sur main :(

Je préfèrerais voir une PR en désactivant les règles de merge plutôt qu'un push sur main sans MR.

Merci et bonne fin de labo.

**Reply by Syseria** [reply]
D'accord merci beaucoup, je vais donc demander à mon binôme de retirer la règle et faire la PR.

Une bonne soirée

**Reply by ludelafo** [reply]
Je suis d'accord avec Hadrien, faites le nécessaire et pas de soucis pour cette fois.

Bonne soirée et bonne semaine !
